cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py

```diff
@@ -791,6 +854,22 @@ def configure_master_services():
     api_opts.add('insecure-bind-address', '127.0.0.1')
     api_opts.add('insecure-port', '8080')
     api_opts.add('storage-backend', 'etcd2')  # FIXME: add etcd3 support
+
+    # TODO: investigate if it's possible to use config file to store args
+    # Handle api-extra-args config option
+    to_add, to_remove = get_config_args()
+    for arg in to_add:
+        hookenv.log('Adding option: {} {}'.format(arg[0], arg[1]))
+        # Make sure old value is gone
+        api_opts.destroy(arg[0])
+        api_opts.add(arg[0], arg[1])
+    for arg in to_remove:
+        hookenv.log('Removing option: {}'.format(arg))
+        api_opts.destroy(arg)
+        # We need to "unset" options by settig their value to "null" string
+        cmd = ['snap', 'set', 'kube-apiserver', '{}=null'.format(arg)]
+        check_call(cmd)
```

Comment:
This problem is basically unsolvable. FlagManger supports options like this:
`{'foo': ['bar', 'baz']}`

Which will render to:
`--foo=bar --foo=baz`

Let's assume an environment has the following state:
`{'foo': ['bar']}`

And we run:
`juju config kubernetes-master api-extra-args="foo=baz"`

There is no way to tell if the operator wants end result to be:
`{'foo': ['bar', 'baz']}`
or
`{'foo': ['baz']}`

My feeling is that we should make it `{'foo': ['baz']}` and trust that the operator knows what they are doing.
Similarly if an operator sets and then unsets certain option it's not obvious that they want it reverted to a value harcoded in the charm. An example might be "--min-request-timeout" which the charm sets to 300s.
The operator might set it to 600s, realize it's too short and "unset" to revert expecting the flag to be removed (and in consequence having k8s default value of 1800s).

ACK your point about positioning above admission-control, I'll get it fixed